lnwallet/chanfunding/wallet_assembler.go

```diff
@@ -285,16 +285,41 @@ func (w *WalletAssembler) ProvisionChannel(r *Request) (Intent, error) {
 		// If there's no funding amount at all (receiving an inbound
 		// single funder request), then we don't need to perform any
 		// coin selection at all.
-		case r.LocalAmt == 0:
+		case r.LocalAmt == 0 && r.FundUpToMaxAmt == 0 &&
+			r.FundMinAmt == 0:
 			break
 
+		// In case this request uses funding up to some maximum amount,
+		// we will call the specialized method for that.
+		case r.FundUpToMaxAmt != 0 && r.FundMinAmt != 0:
+			dustLimit := w.cfg.DustLimit
+			selectedCoins, localContributionAmt, changeAmt,
+				err = CoinSelectUpToAmount(
+				r.FeeRate, r.FundUpToMaxAmt, dustLimit, coins,
+			)
+			if err != nil {
+				return err
+			}
+
+			// In case the selected amount is lower than FundMinAmt
+			// we must return an error.
+			// The FundMinAmt is determined upstream and denotes
+			// either the minimum viable channel size or an amount
+			// sufficient to cover for the initial remote balance.
+			if localContributionAmt < r.FundMinAmt {
```

Comment:
👍

Reply:
another option would be to let the `CoinSelectUpToAmount` function take `minFundAmt` as a parameter and then do this check in there?  just a thought.

Reply:
since it already has a `maxAmount` parameter

Reply:
I still have problems understanding the second part of the comment, minimum viable channel size I understand, but the initial remote balance is still hard to grasp for me. Is this the check that when pushing amount to remote when opening the channel, our local balance needs to account for ?

Reply:
@ellemouton yes that's actually cleaner and I'll change it. One thing though is that now `CoinSelectUpToAmount` depends on the minimum channel size so the function can't be used outside the context of a channel funding flow.

@ziggie1984 yes that's right, the local funding amount needs to account for remote initial balance we push to the remote. If that amount is greater than the minimum funding amount then we have to at least allocate the 'remote initial balance' as local amount.